pages/agent/v3/signed_pipelines.md

```diff
@@ -0,0 +1,162 @@
+# Signed pipelines
+
+> 🚧 Experimental feature
+> Signed pipelines are experimental, and these docs (and the commands and arguments referenced in them) are subject to change. This is also an opt-in feature, so if you'd like to try using signed pipelines, reach out to [support](https://buildkite.com/support).
+
+Signed pipelines are a security feature where pipelines are cryptographically signed when uploaded to Buildkite. Agents then verify the signature before running the job. If an agent detects a signature mismatch, it'll refuse to run the job.
+
+The signature guarantees the origin of jobs by asserting:
+
+- The jobs were uploaded from a trusted source.
+- The jobs haven't been modified after upload.
+
+These signatures mean that if a threat actor could modify a job in flight, the agent would refuse to run it due to mismatched signatures.
+
+<details>
+  <summary>🤔 I think I've seen this before...</summary>
+  <p>This work is inspired by the <a href="https://github.com/buildkite/buildkite-signed-pipeline"><code>buildkite-signed-pipeline</code></a> tool, which you could add to your agent instances. It had a similar idea—signing steps before they're uploaded to Buildkite, then verifying them when they're run. However, it had some limitations, including:</p>
+  <ul>
+    <li>It had to be installed on every agent instance, leading to more configuration.</li>
+    <li>It only supported symmetric HS256 signatures, meaning that every verifier could also sign uploads.</li>
+    <li>It couldn't sign <a href="/docs/pipelines/build-matrix">matrix steps</a>.</li>
+  </ul>
+  <p>This newer version of pipeline signing is built right into the agent and addresses all of these limitations. Being built into the agent, it's also easier to configure and use.</p>
+  <p>Many thanks to <a href="seek.com.au">SEEK</a>, who we collaborated with on the older version of the tool, and whose prior art has been instrumental in the development of this newer version.</p>
+</details>
+
+## Pipeline signatures
+
+Pipeline signatures establish that important aspects of steps haven't been changed since they were uploaded.
+
+The following fields are included in the signature for each step:
+
+- **Commands.**
+- **Environment variables defined in the pipeline YAML.** Environment variables set by the agent, hooks, or the user's shell are not signed and can override the environment a step's command is started with.
+- **Plugins and plugin configuration.**
+- **Matrix configuration.** The matrix configuration is signed as a whole rather than each individual matrix job. This means the signature is the same for each job in the matrix. When signatures are verified for matrix jobs, the agent double-checks that the job it received is a valid construction of the matrix and that the signature matches the matrix configuration.
+- **The step key.** If there is a step key present. [WIP]
```

Comment:
Still want to check what WIP means here

Reply:
in this case it's "we're working on it and we expect to merge it prior to this PR, but don't want to forget to add it to the docs"